include/nbl/builtin/hlsl/central_limit_blur/box_blur.hlsl

```diff
@@ -0,0 +1,297 @@
+
+	type_t operator()( NBL_CONST_REF_ARG( type_t ) value )
+	{
+		type_t retval = nbl::hlsl::workgroup::inclusive_scan<bin_op_t, arithmeticSz>::template __call<ScratchProxy<type_t, 0> >( value, prefixSumsAccessor );
+		// we barrier before because we alias the accessors for Binop
+		prefixSumsAccessor.workgroupExecutionAndMemoryBarrier();
+		return retval;
+	}
+};
+
+prefix_sum_t workgroupPrefixSum;
+
+float32_t workgroupBroadcast( NBL_CONST_REF_ARG( float32_t ) value, uint32_t index )
+{
+	using accessor_t = ScratchProxy<float32_t, arithmeticSz>;
+	return nbl::hlsl::workgroup::Broadcast<float32_t, accessor_t>( value, broadcastAccessor, index );
+}
+
+// Todo: This spillage calculation is hacky! The lower bound of `_NBL_GLSL_EXT_BLUR_SPILLAGE_LOWER_BOUND_` is just an adhoc
+// thing which happens to work for all images until now but it could very well break for the next image.
+// This:
+// `#define LOCAL_SPILLAGE ITEMS_PER_THREAD`
+// seems to be a safe lower bound, however, it must be inefficient.
+static const uint32_t SPILLAGE_LOWER_BOUND = ( WORKGROUP_SIZE / 4 );
+
+static const uint32_t IMPL_LOCAL_SPILLAGE = ( ( arithmeticSz - 1 ) / WORKGROUP_SIZE + 1 );
+
+static const uint32_t LOCAL_SPILLAGE =
+( IMPL_LOCAL_SPILLAGE > WORKGROUP_SIZE ) ? WORKGROUP_SIZE :
+	( ( IMPL_LOCAL_SPILLAGE < WORKGROUP_SIZE ) ? SPILLAGE_LOWER_BOUND : IMPL_LOCAL_SPILLAGE );
+
+
+uint32_t DataIndex( uint32_t i ) // ?
+{
+	return ( i * gl_WorkGroupSize().x ) + nbl::hlsl::glsl::gl_LocalInvocationIndex();
+}
+
+//TODO: glsl's mod(x, y) = x - y * floor(x/y). Would prolly need to implement a glsl mod
+void BoxBlur(
+	const uint32_t ch,
+	const float32_t inRadius,
+	const uint32_t wrapMode,
+	const float32_t4 borderColor,
+	BufferAccessor textureAccessor
+) {
+	float32_t blurred = textureAccessor.get( DataIndex( i ), ch );
+
+	for( uint32_t pass = 0u; pass < PASSES_PER_AXIS; ++pass )
+	{
+		float32_t previousBlockSum = 0.f;
+
+		float32_t spill[ LOCAL_SPILLAGE ];
+		for( uint32_t i = 0u; i < LOCAL_SPILLAGE; ++i )
+		{
+			float32_t scanResult = workgroupPrefixSum( blurred ) + previousBlockSum;
+			spill[ i ] = scanResult;
+			previousBlockSum = workgroupBroadcast( spill[ i ], gl_WorkGroupSize().x - 1u );
+		}
+
+		float32_t scanResult = workgroupPrefixSum( blurred ) + previousBlockSum;
+		previousBlockSum = workgroupBroadcast( scanResult, gl_WorkGroupSize().x - 1u );
+
+		uint32_t idx = DataIndex( i ); // ?
+		prefixSumsAccessor.set( idx, scanResult );
+
+		for( uint i = 0u; i < LOCAL_SPILLAGE; ++i )
+		{
+			uint32_t idx = DataIndex( i ); // ?
+			prefixSumsAccessor.set( idx, spill[ i ] );
+		}
+		prefixSumsAccessor.workgroupExecutionAndMemoryBarrier();
+
+		uint32_t idx = DataIndex( i );
+
+		if( idx < AXIS_DIM )
+		{
+			const float32_t radius = inRadius * AXIS_DIM;
+			float32_t left = float32_t( idx ) - radius - 1.f;
+			float32_t right = float32_t( idx ) + radius;
+			const uint32_t last = AXIS_DIM - 1;
+
+			float32_t result = 0.f;
+
+			if( right <= last )
```

Comment:
this check is wrong, it should check if `right` is still within the texture/buffer, so you need `ceil(right)<float(itemCount)`

You could `assert(itemCount<MaxItemCount);` if the assert worked at all